### ISSUE TITLE
DT::datatable in main column in Tufte html output

### ISSUE THREAD

**Issue by randrescastaneda** [rstudio/DT#762]
Dear all, 

I need to place a `DT::datatable` in the main column of a Tufte html document, but everything that I've tried has failed. A few days ago I posted the issue in [Stack Overflow](https://stackoverflow.com/questions/59913042/dtdatatable-in-main-column-in-tufte-html-output), but I have not received one single answer. 

I would highly appreciate any help. You may find all the details in the [Stack Overflow](https://stackoverflow.com/questions/59913042/dtdatatable-in-main-column-in-tufte-html-output) question. 

Thanks. 
Andres. 
---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.name/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('DT')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/DT')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by shrektan** [comment]
I'm not sure if it's the best way but it works. Just put the below code in your Rmd file:

```html
<style type = 'text/css'>
div.datatables {
  width: 55% !important;
}
</style>
```

## The full example code:

``````rmd
---
title: "Tufte Handout"
subtitle: "An implementation in R Markdown"
author: "JJ Allaire and Yihui Xie"
date: "`r Sys.Date()`"
output:
  tufte::tufte_html: default
  tufte::tufte_handout:
    citation_package: natbib
    latex_engine: xelatex
  tufte::tufte_book:
    citation_package: natbib
    latex_engine: xelatex
link-citations: yes
---

<style type = 'text/css'>
div.datatables {
  width: 55% !important;
}
</style>

```{r setup, include=FALSE}
library(tufte)
# invalidate cache when the tufte version changes
knitr::opts_chunk$set(tidy = FALSE, cache.extra = packageVersion('tufte'))
options(htmltools.dir.version = FALSE)
```

```{r}
plot(1:10)
```

```{r high-range-cv, echo = FALSE}
DT::datatable(iris)
```
``````

**Comment by shrektan** [comment]
Well, it looks a bug to me... because leaflet will respect the `out.width` argument. I'm investigating...



``````rmd
---
title: "out.width doesn't work on DT"
---

```{r out.width='50%', echo=FALSE}
library(leaflet)
leaflet() %>% addTiles() %>%
  setView(-93.65, 42.0285, zoom = 17) %>%
  addPopups(
    -93.65, 42.0285,
    'Here is the <b>Department of Statistics</b>, ISU'
  )
library(DT)
datatable(iris)
```
``````

<img width="1023" alt="image" src="https://user-images.githubusercontent.com/8368933/73457384-8f8d6400-43ae-11ea-89d5-d37f4afede5c.png">

**Comment by shrektan** [comment]
Oh, I see the cause is the `knitr.figure = FALSE` in

https://github.com/rstudio/DT/blob/e0de133f01d01ca1ed11d2573096e7fa8253611f/R/datatables.R#L291-L296

and the change was made by #298. I don't think I understand the benefits of setting this argument to `FALSE`.

 @jjallaire @yihui  Is the reason mentioned in #298 still valid after four years? Sometimes, it frustrates the users, in my opinion.

**Comment by randrescastaneda** [comment]
Thank you so much for digging into the problem, @shrektan . I highly appreciate it. So, is there something I can do while this problem is solved at "official" level?

Thanks. 
Anddres

**Comment by shrektan** [comment]
@randrescastaneda please see this comment https://github.com/rstudio/DT/issues/762#issuecomment-580250719

> You can set the `width` argument in `DT::datatable()` (better with `options = list(scrollX = TRUE)` because when the table is too wide you may want a x-scroller.
> 
> ## The full example code:
> ```gfm
> ---
> title: "Tufte Handout"
> subtitle: "An implementation in R Markdown"
> author: "JJ Allaire and Yihui Xie"
> date: "`r Sys.Date()`"
> output:
>   tufte::tufte_html: default
>   tufte::tufte_handout:
>     citation_package: natbib
>     latex_engine: xelatex
>   tufte::tufte_book:
>     citation_package: natbib
>     latex_engine: xelatex
> link-citations: yes
> ---
> 
> ```{r setup, include=FALSE}
> library(tufte)
> # invalidate cache when the tufte version changes
> knitr::opts_chunk$set(tidy = FALSE, cache.extra = packageVersion('tufte'))
> options(htmltools.dir.version = FALSE)
> ```
> 
> ```{r}
> plot(1:10)
> ```
> 
> ```{r high-range-cv, echo = FALSE}
> DT::datatable(iris, width = '55%', options = list(scrollX = TRUE))
> ```
> ```

**Comment by randrescastaneda** [comment]
Hi @shrektan , You're absolutely right. I apologize for the confusion. Yes, your solution worked very well. 

Thank you so much. 
Best,

**Comment by yihui** [comment]
@shrektan To be honest, I don't remember (perhaps I never truly understood #298). I agree that it is annoying when the user setting `out.width` is ignored.

**Comment by shrektan** [comment]
One possible reason is that, in pagination mode, the height of a DT table is determined by the content and can't be fixed by the user. If it's somehow set too small, the table will overlap on the next HTML content. 

``````r
---
title: "Vignette Title"
author: "Vignette Author"
date: "`r Sys.Date()`"
output: html_document
---

```{r, out.width = "100%", out.height = "50px"}
DT::datatable(iris)
```

```{r}
DT::datatable(iris)
```
``````

### knitr.figure = FALSE

<img width="974" alt="image" src="https://user-images.githubusercontent.com/8368933/73875082-9f65e600-488f-11ea-8372-240ee84ed7dc.png">


### knitr.figure = TRUE

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/8368933/73874908-4c8c2e80-488f-11ea-9877-dcfd24e0e3e4.png">

**Comment by jjallaire** [comment]
Yes, typically DT output doesn't gracefully handle being put in a fixed
container size. There are some special adaptations we made for
flexdashboard however these aren't the default for a normal Rmd.

**Comment by brandonerose** [comment]
I was having an issue with a DT::datatable that was too wide for my flex dashboard. I made a fork of DT and changed knitr.figure to TRUE, which did not work for me. Adding the following chunk via Joe Cheng corrected my problem...

```{css}
.chart-wrapper {
  overflow-x: scroll;
}
```
